### ISSUE TITLE
How to get the output of a layer in the middle of a neural network?

### ISSUE THREAD

**Issue by chang-change** [lululxvi/deepxde#607]
Dear Dr Lu Lu,
For the two physical quantities A and B, the result of A will have a direct impact on B. So, I want to build a neural network like this, A is the output of a certain layer in the middle of the neural network, and B is the output of the last layer of the neural network. However, I don't know how can I get the value of A? Is there a better way to deal with this clearly causal variable in DeepXDE?
Looking forward to your reply, thank you!

**Comment by praksharma** [comment]
May be you can obtain the weights here https://deepxde.readthedocs.io/en/latest/_modules/deepxde/model.html#LossHistory.set_loss_weights

Then simply do a forward propagation manually. May be there is a better approach. This one should work.

**Comment by chang-change** [comment]
@praksharma  Thanks for your response. I want to get the derivative of the intermediate variable, then construct the residual equation. Finally, adds the residual equation to the loss.

**Comment by chang-change** [comment]
![2022-04-12_110111](https://user-images.githubusercontent.com/45302365/162870802-c922d438-d3ec-4eff-974d-755bba0a2f27.png)

**Comment by tsarikahin** [comment]
I think this is not supported by DeepXDE. You have to create your own architecture.

**Comment by praksharma** [comment]
> ![2022-04-12_110111](https://user-images.githubusercontent.com/45302365/162870802-c922d438-d3ec-4eff-974d-755bba0a2f27.png)

I don't think this structure is robust for 2D and 3D problems. Did they publish this paper? By the way, this architecture is simple and you can write a code in 1 hour.

**Comment by chang-change** [comment]
@tsarikahin @praksharma  Thanks for your reply. I know I can create the network by myself. I just want to know if  the Deepxde can do it. The Deepxde  is a magic tool, and it can save my time. By Deepxde, I can define geometry, boundary conditions, and training data, etc. It's convenient. But, I don't know that how to combine Deepxde with my network. Can you give me some suggestions? Thank you very much.

**Comment by lululxvi** [comment]
@praksharma @tsarikahin Good comments. 

Actually it is doable in DeepXDE, although not in a straightforward way. See FAQ Q: A standard network doesn’t work for me, and I want to implement a network with a special structure/property.

**Comment by chang-change** [comment]
@lululxvi Thanks for your reply.
I use `apply_feature_transform` to transform the inputs of  network to the feature I need. The code as follows.
`def input_feature_transform(x):
    xx = x[:,0:1]
    yy = x[:,1:2]
    zeta  = xx*(xx**2+yy**2+1)/(xx**2+yy**2)
    eta =  yy*(xx**2+yy**2-1)/(xx**2+yy**2)
    inputs = tf.concat([zeta,eta],axis = 1)
    return inputs
net.apply_feature_transform(input_feature_transform)
`
How can I get the derivates of output of the network with respect to the features, instead of the derivates of output of the network with respect to spatial-time coordinate. Can you give me some suggestions? Thanks!

**Comment by lululxvi** [comment]
It is not straightforward to do. Then why don't you directly use `zeta` and `eta` as the network input?

**Comment by chang-change** [comment]
@lululxvi  Thanks for your reply. I want to use Conformal Mapping to transform the problem of flowing around a cylinder to the problem of flowing around the wing. Because I don't know how to define the wing geometry by DeepXDE. Can you give me some advice?  Thank you. 
![保角映射](https://user-images.githubusercontent.com/45302365/163999057-990b109f-25fe-4ba7-a02b-c506fb8d6070.png)

**Comment by lululxvi** [comment]
A few ideas:
- Use a polygon to approximate the geometry;
- Still use x and y as the input, but converting the derivative wrt zeta to the derivative wrt x and y
- Modify the source code so that you can compute the derivative wrt zeta

**Comment by chang-change** [comment]
@lululxvi Thanks for your reply. How can I use polygon  to approximate the irregular shape?

**Comment by lululxvi** [comment]
See https://github.com/lululxvi/deepxde/issues/634
